File: marketdata-common/src/main/java/com/handson/dto/Stock.java
package com.handson.dto;

import com.google.gson.Gson;

public class Stock {

    public final String code;

    public final String companyName;

    public final String market;

    public Stock(String code, String companyName, String market) {
        this.code = code;
        this.companyName = companyName;
        this.market = market;
    }

    public static Stock fromJson(String input) {
        return new Gson().fromJson(input, Stock.class);
    }


    public String toJson() {
        return new Gson().toJson(this);
    }

    @Override
    public String toString() {
        return toJson();
    }
}

File: CSharp/NetCore/MarketData/src/MarketDataCommon/Dto/Stock.cs
using Newtonsoft.Json;

namespace MarketDataCommon.Dto
{
    public class Stock
    {
        public string Code { get; set; }
        public string CompanyName { get; set; }
        public string Market { get; set; }

        public Stock(string code, string companyName, string market)
        {
            Code = code;
            CompanyName = companyName;
            Market = market;
        }

        public static Stock FromJson(string input)
        {
            return JsonConvert.DeserializeObject<Stock>(input);
        }

        public string ToJson()
        {
            return JsonConvert.SerializeObject(this);
        }

        public override string ToString() => ToJson();
    }
}
